src/app/qgsattributetabledialog.cpp

```diff
@@ -465,6 +486,47 @@ void QgsAttributeTableDialog::columnBoxInit()
   }
 }
 
+void QgsAttributeTableDialog::storedFilterExpressionBoxInit()
+{
+  const auto constActions = mStoredFilterExpressionMenu->actions();
+  for ( QAction *a : constActions )
+  {
+    mStoredFilterExpressionMenu->removeAction( a );
+    delete a;
+  }
+
+  const QList< QgsStoredExpression > storedExpressions = mLayer->storedExpressionManager()->storedExpressions();
+  for ( const QgsStoredExpression &storedExpression : storedExpressions )
+  {
+    QAction *storedExpressionAction = new QAction( storedExpression.name, mFilterButton );
+    connect( storedExpressionAction, &QAction::triggered, this, [ = ]()
+    {
+      setFilterExpression( storedExpression.expression, QgsAttributeForm::ReplaceFilter, true );
+    } );
+    mStoredFilterExpressionMenu->addAction( storedExpressionAction );
+  }
+}
+
+void QgsAttributeTableDialog::storeExpressionButtonInit()
+{
+  if ( mActionHandleStoreFilterExpression->isChecked() )
+  {
+    mActionHandleStoreFilterExpression->setToolTip( tr( "Delete stored expression" ) );
+    mActionHandleStoreFilterExpression->setText( tr( "Delete Stored Expression" ) );
```

Comment:
Does this show a confirmation prompt? If so, the text should end with "..."

Reply:
It does not show a confirmation prompt. Should it? I decided not to have one to be quick in adding/removing the stored expressions like bookmarks and because of the low importance...